src/runtime_src/xdp/profile/plugin/aie_profile/aie_plugin.cpp

```diff
@@ -505,6 +505,22 @@ namespace xdp {
           //else if (startEvents.at(i) == XAIE_EVENT_GROUP_ERRORS_MEM)
           //  XAie_EventGroupControl(aieDevInst, loc, mod, startEvents.at(i), GROUP_ERROR_MASK);
 
+          // Enable floating point exception events
```

Comment:
AIEProfilingPlugin::setMetrics() is about 250 lines long. Please break up this function into logical pieces.